pygeos/lib/__init__.py

```diff
@@ -0,0 +1,3 @@
+from pygeos.lib.core import *
```

Comment:
This is so that we didn't need to alter the imports from `pygeos.lib` as part of migrating `pygeos.lib` the module to `pygeos.lib` the package; otherwise we'd have to update a number of the Python files with longer import paths.

The other idea behind this package is that there is no meaningful distinction between C extension and Cython extensions when using them at the Python level.

However, this also leads to circular imports at build time, so it needs a bit more work.